### PR TITLE
EVG-2848 Refactor checkTask and checkHost for use in generate.tasks

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: model/validate.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+)
+
+const (
+	ApiTaskKey       apiTaskKey       = 0
+	ApiHostKey       apiHostKey       = 0
+	ApiProjectKey    apiProjectKey    = 0
+	ApiProjectRefKey apiProjectRefKey = 0
+)
+
+func ValidateTask(taskId string, checkSecret bool, r *http.Request) (*task.Task, int, error) {
```
Comment (unrealsolver) [inline model/validate.go]: Could you please add a doc string?

File: model/validate.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		secret := r.Header.Get(evergreen.TaskSecretHeader)
+		if secret != t.Secret {
+			return nil, http.StatusConflict, errors.Errorf("Wrong secret sent for task %s: Expected %s but got %s",
+				taskId, t.Secret, secret)
+		}
+	}
+	return t, http.StatusOK, nil
+}
+
+func ValidateHost(hostId string, r *http.Request) (*host.Host, int, error) {
```
Comment (unrealsolver) [inline model/validate.go]: Could you please add a doc string?

File: model/validate.go (hunk 1 above)
Comment (bsamek) [reply model/validate.go]: done

File: model/validate.go (hunk 2 above)
Comment (bsamek) [reply model/validate.go]: done

File: rest/route/generate_routes.go [hunk 3]
```diff
@@ -38,20 +40,36 @@ func (h *generateHandler) Handler() RequestHandler {
...
+	if _, code, err := dbModel.ValidateTask(h.taskID, true, r); err != nil {
+		return &rest.APIError{
+			StatusCode: code,
+			Message:    "task is invalid",
+		}
+	}
+	if _, code, err := dbModel.ValidateHost("", r); err != nil {
+		return &rest.APIError{
+			StatusCode: code,
+			Message:    "task is invalid",
```
Comment (john-m-liu) [inline rest/route/generate_routes.go]: "host is invalid"

**Pull request merged**